index.js

```diff
@@ -45,11 +45,13 @@ class BindingComponent extends React.Component {
     }
 
     this.props.bindStateCtx.setState((prevState, currentProps) => {
+      let path;
       let pathReversed;
       let overlay = val;
       let pathIndex = 0;
       try {
-        pathReversed = currentProps.bindStateName.split('.').reverse();
+        path = currentProps.bindStateName.split('.');
```

Comment:
The variable `path` seems to be live in this `try` block, so that'd better:

```js
try {
  let path = currentProps.bindStateName.split('.'); // don't use const is because .reverse() is a in-place function.
  ...
}
```